src/realm/object-store/shared_realm.cpp

```diff
@@ -641,6 +641,16 @@ util::Optional<VersionID> Realm::current_transaction_version() const
     return ret;
 }
 
+// Get the version of the current snapshot
+util::Optional<DB::version_type> Realm::current_transaction_snapshot_version() const
```

Comment:
This function's name is inaccurate. It returns the _latest_ version, not the version of the current transaction.